atomic_reactor/plugins/exit_pulp_publish.py

```diff
@@ -68,10 +69,23 @@ def __init__(self, tasker, workflow, pulp_registry_name,
         # call parent constructor
         super(PulpPublishPlugin, self).__init__(tasker, workflow)
         self.workflow = workflow
-        self.pulp_handler = PulpHandler(self.workflow, pulp_registry_name, self.log,
-                                        pulp_secret_path=pulp_secret_path,
-                                        username=username, password=password,
-                                        dockpulp_loglevel=dockpulp_loglevel)
+        self.pulp_registry_name = pulp_registry_name
+        self.username = username
+        self.password = password
+        self.pulp_secret_path = pulp_secret_path
+        self.dockpulp_loglevel = dockpulp_loglevel
+        self.pulp_handler = self._get_pulp_session()
+
+    def _get_pulp_session(self):
+        try:
+            return get_pulp_session(self.workflow, self.log, self.dockpulp_loglevel)
+        except KeyError:
+            pass
+
+        return PulpHandler(self.workflow, self.pulp_registry_name, self.log,
+                           pulp_secret_path=self.pulp_secret_path,
+                           username=self.username, password=self.password,
+                           dockpulp_loglevel=self.dockpulp_loglevel)
```

Comment:
This code is repeated in several other plugins. It should be refactored and moved into pulp_util.py